scripts/transcriptomic_data/get_transcriptomic_data.py

```diff
@@ -0,0 +1,156 @@
+    # description is pulled from ENA BioSample
+    # if the description does not exist, the sample accession is provided for tracibility
+
+    # try:
+    #    description = biosample_data['characteristics']['description'][0]['text']
+    # except KeyError:
+    #    description = accession
+
+    if "characteristics" in biosample_data and "description" in biosample_data["characteristics"]:
+        description = biosample_data["characteristics"]["description"][0]["text"]
+    else:
+        description = accession
+
+    # we will likely come across more characters that should be removed from sample names
+    # improvements on this should come with the sample names project
+
+    replace_chars = "()/\\"
+    sample = "".join(
+        map(
+            lambda char: "_" if char == " " else "",
+            (sample.replace(char, "") for char in replace_chars),
+        )
+    )
+    return (sample, description)
+
+
+def get_data_from_ena(# pylint: disable=too-many-locals
+    taxon_id: int, read_type: str
+) -> List[str]:
+    """Query ENA API to get short or long read data"""
+    csv_data = []
+    
+    query = f"tax_eq({taxon_id})"
+    if read_type == "short":
+        query += " AND instrument_platform=ILLUMINA AND library_layout=PAIRED"
+    else:
+        query += " AND (instrument_platform=OXFORD_NANOPORE OR instrument_platform=PACBIO_SMRT)"
+
+    query += " AND library_source=TRANSCRIPTOMIC"
+
+    search_url = f"https://www.ebi.ac.uk/ena/portal/api/search?display=report&query={query}&domain=read&result=read_run&fields=sample_accession,run_accession,fastq_ftp,read_count,instrument_platform"  # pylint: disable=line-too-long
+    search_result = requests.get(search_url)
+    results = search_result.text.strip().split("\n")[1:]
+
+    # fields for the csv that we don't need any more but could break
+    # downstream things so we'll keep them for now
+    is_paired = "1"  # we always used paire-end data
+    is_mate_1 = "-1"
+    # read_length is always incorrect when calculated from ENA and is not needed with STAR anyway
+    read_length = "1"
+    is_plus_13 = "0"  # outdated sequencing info
+    centre = "ENA"  # we get everything from ENA anyway
+
+    for row in results:
+        row_data = row.split()
+        run_accession = row_data[0]
+        sample_accession = row_data[1]
+        sample, description = get_sample_info(sample_accession)
+        read_count = int(row_data[3])
+        instrument_platform = row_data[4]
+        for file in row_data[2].split(";"):
+            file_path=os.path.basename(file)
+            csv_data.append(
+                (
+                sample,
+                run_accession,
+                is_paired,
+                file_path,
+                is_mate_1,
+                read_length,
+                is_plus_13,
+                centre,
+                instrument_platform,
+                description,
+                )
+            )
+    return csv_data
+
+class text:
```

Comment:
Is this needed in this script or is it just leftover from a copy-and-paste from the other script? It doesn't matter, just for future cleaning, I guess

Reply:
It is not needed here, I just removed it in the new commit